### ISSUE TITLE
Spyder 4.0.0 and dev hangs when restarting gnome-shell

### ISSUE THREAD

**Issue by oscargus** [spyder-ide/spyder#11117]
### Issue Report Checklist

* [x] Searched the [issues page](https://github.com/spyder-ide/spyder/issues?q=is%3Aissue) for similar reports
* [x] Read the relevant sections of the [Spyder Troubleshooting Guide](https://github.com/spyder-ide/spyder/wiki/Troubleshooting-Guide-and-FAQ) and followed its advice
* [x] Reproduced the issue after updating with ``conda update spyder`` (or ``pip``, if not using Anaconda)
* [x] Tried basic troubleshooting (if a bug/error)
    * [x] Restarted Spyder
    * [ ] Reset preferences with ``spyder --reset``
    * [ ] Reinstalled the latest version of [Anaconda](https://www.anaconda.com/download/)
    * [x] Tried the other applicable steps from the Troubleshooting Guide
* [x] Completed the **Problem Description**, **Steps to Reproduce** and **Version** sections below

## Problem Description
When restarting gnome-shell (at least on my CentOS 7 machine) Spyder hangs (window does not update anymore, no responses, and OS reports that it is not responding). This holds for both the current master and the latest version in conda.

However, when running with debug (`bootstrap.py --debug` or `--debug-info verbose`) it doesn't hang so I have really no clue what is happening.

This may seem like a rare case. However, as gnome-shell tends to grow quite big in size, a rather common approach to deal with this is to restart it using a cron job. At least that is what my university does.

### What steps reproduce the problem?

1. Start spyder on a machine using Gnome
2. Press Alt+F2 to get the Gnome command box
3. Enter `r` to restart

### What is the expected output? What do you see instead?
No hang. Still be able to work with Spyder.

## Versions

* Spyder version: 4.0.0/5.0.0 dev
* Python version: 3.7
* Qt version: 5.9.6 
* PyQt version: 5.9.2
* Operating System name/version: CentOS 7 Linux 3.10.0

### Dependencies

Slightly different depending on version, but all are met.

**Comment by ccordoba12** [comment]
> When restarting gnome-shell (at least on my CentOS 7 machine) Spyder hangs (window does not update anymore, no responses, and OS reports that it is not responding)

Thanks for reporting. This is very bad indeed!

Unfortunately this is going to take us some time to solve because we need to install a virtual machine with CentOS 7 and the gnome-shell version that comes with it to reproduce it and understand how to fix it.

> However, as gnome-shell tends to grow quite big in size

You mean it starts to consume too much RAM?

**Comment by goanpeca** [comment]
I can work on this @ccordoba12 I have a centos7 vm installed

**Comment by ccordoba12** [comment]
Great! Thanks @goanpeca!

**Comment by oscargus** [comment]
@ccordoba12 Yes, it seem that gnome-shell sometimes consumes much resident RAM, especially combined with NVidia graphics cards. Most of it may have been fixed recently it seems, but the current solution of our university (since we haven't upgraded Gnome version) is to every half hour check if it has reached 400 MB resident size and in that case reset it. Which it does quite often for me... Almost every half hour...

Anyway, I tried on Ubuntu now (Spyder 4.0.0) with Gnome and the problem in not there at least.

That lead me to dig a bit more into it and it turns out that there are more things required to trigger the bug. I have three screens, portrait - landscape - portrait, and I have had issues that gnome-shell doesn't start with the correct orientation when it starts and restarts. It turns out that it is only when I have rotated screens and they are rotated back on a restart that in hangs. If I do not have any rotation, restarting gnome-shell works fine and Spyder continues to work properly. I also tested on a colleagues computer and there it doesn't happen, neither the automatic back-rotation or Spyder hanging. The main difference seems to be that I (probably) have a dedicate Nvidia driver installed (since I have specific Nvidia configuration tools), while he also has an Nvidia card, but a simpler version and it seems like no configuration tools installed and therefore also (I believe) a different driver.

So it seems like that one will need to have rotated screens and specific Nvidia drivers (and a bit of bad luck) for it to happen. OS or not is hard to say since I only have CentOS installed on my machine.

Anyway, after finally figuring out what caused the automatic rotations last week, I've convinced them to turn the automatic restarts off. Hard to say if it is even possible to figure out the bug (or if there even is a bug in Spyder), but as long as it doesn't restart it seems to work properly and when it restarts it is only under certain conditions that are quite unlikely to get. So although it is a bit of a bad bug, it is quite rare one could say.

(I have a Quadro P2000 graphics card now and earlier a 1050. I do not know if Spyder hangs on the 1050, but it had the same rotation issue, so I guess it did. I only recently upgrade Anaconda to Spyder 4.0.0 and ran from source with debug most of the time earlier.)
